matrix-hookshot: fails to execute, typescript app not built

### Nixpkgs version

- Stable (25.11)

### Describe the bug

matrix-hookshot does fail on start. It tries to execute `…/lib/node_modules/matrix-hookshot/lib/App/BridgeApp.js` which does not exist.

Only the source typescript files are present:
`…/lib/node_modules/matrix-hookshot/src/App/BridgeApp.ts`

It looks like the typesrcipt app is not being built.

### Steps to reproduce

```
nix build nixpkgs#matrix-hookshot`
./result/bin/matrix-hookshot
```

### Expected behaviour

It should not fail

### Screenshots

_No response_

### Relevant log output

```console
node:internal/modules/cjs/loader:1386
  throw err;
  ^

Error: Cannot find module '/nix/store/jcj6jz67g8m5y91mb7rppxbx0yk9aamd-matrix-hookshot-7.1.0/lib/node_modules/matrix-hookshot/lib/App/BridgeApp.js'
    at Function._resolveFilename (node:internal/modules/cjs/loader:1383:15)
    at defaultResolveImpl (node:internal/modules/cjs/loader:1025:19)
    at resolveForCJSWithHooks (node:internal/modules/cjs/loader:1030:22)
    at Function._load (node:internal/modules/cjs/loader:1192:37)
    at TracingChannel.traceSync (node:diagnostics_channel:328:14)
    at wrapModuleLoad (node:internal/modules/cjs/loader:237:24)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:171:5)
    at node:internal/main/run_main_module:36:49 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}

Node.js v22.21.1
```

### Additional context

_No response_

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.12.61, NixOS, 26.05 (Yarara), 26.05pre-git`
 - multi-user?: `no`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.31.2+1`
 - channels(root): `""`
 - channels(christoph): `"home-manager, nixpkgs"`
 - nixpkgs: `not found`

### Notify maintainers

@chvp (maintainer)
@pyrox0 (did the last update)

Many thanks for your time!

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+bug%22+-label%3A%226.topic%3A+darwin%22+-label%3A%226.topic%3A+nixos%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/ma/matrix-hookshot/package.nix
+++ b/pkgs/by-name/ma/matrix-hookshot/package.nix
@@ -6,7 +6,6 @@
   makeWrapper,
   matrix-sdk-crypto-nodejs,
   yarnConfigHook,
-  yarnInstallHook,
   cargo,
   rustPlatform,
   rustc,
@@ -43,7 +42,6 @@ stdenv.mkDerivation (finalAttrs: {
   nativeBuildInputs = [
     rustPlatform.cargoSetupHook
     yarnConfigHook
-    yarnInstallHook
     pkg-config
     cargo
     rustc
@@ -71,9 +69,31 @@ stdenv.mkDerivation (finalAttrs: {
     runHook postBuild
   '';
 
+  installPhase = ''
+    runHook preInstall
+
+    rm -rf ./node_modules
+    yarn install --production --offline --force --ignore-engines --no-bin-links --frozen-lockfile
+
+    # Re-install matrix-sdk-crypto-nodejs
+    rm -rf node_modules/@matrix-org/matrix-sdk-crypto-nodejs
+    cp -r ${matrix-sdk-crypto-nodejs}/lib/node_modules/@matrix-org/matrix-sdk-crypto-nodejs \
+      node_modules/@matrix-org/matrix-sdk-crypto-nodejs
+    chmod -R a+rwx node_modules/@matrix-org/matrix-sdk-crypto-nodejs
+
+    mkdir -p $out/lib/node_modules/matrix-hookshot/
+    mkdir $out/bin
+
+    mv ./lib/* $out/lib/node_modules/matrix-hookshot
+    mv ./public ./assets ./node_modules ./package.json $out/lib/node_modules/matrix-hookshot
+
+    runHook postInstall
+  '';
+
   postInstall = ''
-    makeWrapper '${lib.getExe nodejs}' "$out/bin/matrix-hookshot" --add-flags \
-        "$out/lib/node_modules/matrix-hookshot/lib/App/BridgeApp.js"
+    makeWrapper '${lib.getExe nodejs}' "$out/bin/matrix-hookshot" \
+      --set NODE_ENV "production" \
+      --add-flags "$out/lib/node_modules/matrix-hookshot/App/BridgeApp.js"
   '';
 
   passthru.updateScript = nix-update-script { };

